core/contracts/Voting.sol

```diff
@@ -517,86 +486,44 @@ contract Voting is Testable, MultiRole, OracleInterface, VotingInterface, Encryp
         returns (bool _hasPrice, int price, string memory err)
     {
         PriceRequest storage priceRequest = _getPriceRequest(identifier, time);
-        uint resolutionVotingRound = priceRequest.lastVotingRound;
-        uint lastActiveVotingRound = voteTiming.getLastUpdatedRoundId();
-
-        if (resolutionVotingRound < lastActiveVotingRound) {
-            // Price must have been requested in the past.
-            if (resolutionVotingRound == 0) {
-                return (false, 0, "Price was never requested");
-            }
-
-            // Grab the resolution voting round to compute the resolved price.
-            VoteInstance storage voteInstance = priceRequest.voteInstances[resolutionVotingRound];
-            (, int pastResolvedPrice) = voteInstance.resultComputation.getResolvedPrice(
-                _computeGat(resolutionVotingRound));
+        if (priceRequest.lastVotingRound == 0) {
```

Comment:
We have these if statements in a lot of places - I feel like it would be a bit more readable if we had a method like `_isActive` but that returned an enum that describes the "state" of the price request - something like:
```
NotRequested
Resolved
Active
Future
```
Or something like that.

The issue with this is that it will obviously mean more lines of code and be less efficient, since it'll turn one set of ifs into two (it won't get rid of these ifs, it will just make them more readable). The other issue is that solidity doesn't have switch statements IIUC, which makes the result less readable. I'm not really sold on it. I just feel like it could mean a reader wouldn't have to repeatedly reread these ifs which are fairly obtuse. Thoughts?

Reply:
Yeah I thought about something like that and decided against it, for the reasons that you outlined. I tried typing it out, and I think it actually looks OK. Not perfect, but a net positive for readability. Thoughts?